server/handlers/order.js

```diff
@@ -58,3 +59,27 @@ exports.createOrderHandler = async function (req, res, next) {
 
   return res.json(Order.fromMongoJSON(result));
 };
+
+exports.getOrdersByClientIDValidator = [orderClientIDQueryValidator];
+
+exports.getOrdersByClientID = async function (req, res, next) {
+  let result;
+
+  try {
+    result = await dao.getOrdersByClientID(req.query.clientID);
+  } catch (err) {
+    console.error(
+      `getOrdersByClientID() -> couldn't retrieve client's orders: ${err}`
+    );
+    return res.status(500).end();
+  }
+
+  if (!result.length) {
```

Comment:
Ok I changed my opinion about this case. As I did when I've implemented findClients you can just do not consider this case and return a an empty array. So you could just write return res.json(result.map(element) => Order.fromMongoJSON(element));

After all it's not an error. It shouldn't return 404 but an empty array.